src/components/tables/Table.tsx

```diff
@@ -123,6 +125,12 @@ export class Table extends React.Component<ITableProps, {}> {
     }
   }
 
+  componentWillUpdate(tableCompositeState: any) {
```

Comment:
what you receive from `componentWillUpdate` is not the tableCompositeState but the nextProps as your first argument, i think you should rename it as such https://reactjs.org/docs/react-component.html#componentwillupdate.   
  
```
componentWillUpdate(nextProps: ITableProps) {
//...
}
```